pkg/web/apps/apps_service.go

```diff
@@ -34,3 +35,21 @@ func (a *appsService) GetApps() (*[]models.App, error) {
 
 	return apps, nil
 }
+
+// GetAppByID retrieves app by id from the repository
+func (a *appsService) GetAppByID(id string) (*models.App, error) {
+
+	app, err := a.repository.GetAppByID(id)
+
+	if err != nil {
+		return nil, err
+	}
+	if app == nil {
```

Comment:
If you return the error `models.ErrNotFound` in the repository layer then you won't need to do a check then you won't need this if statement.

Reply:
:+1: